app/abilities/ability.rb

```diff
@@ -52,9 +52,11 @@ def initialize(user)
         can [:show, :edit, :update], :preference
         can [:edit, :update], :profile
         can [:new, :create], Report
-        can [:mine, :new, :create, :edit, :update, :destroy], Trace
+        can [:mine], Trace
         can [:account, :go_public], User
 
+        can [:new, :create, :edit, :update, :destroy], Trace if user.blocks.active.empty?
+
```

Comment:
I would rearrange these to keep the two Trace lines beside each other.

Also, we need to discuss putting the blocking logic into the ability file - it's not done for the existing blocking logic.